source/ui.js

```diff
@@ -214,44 +215,58 @@ const ui = async (options, {pkg, rootDir}) => {
 		}
 	}
 
+	const needsPrereleaseTag = answers => options.runPublish && (answers.version?.isPrerelease() || answers.customVersion?.isPrerelease()) && !options.tag;
+	const canBePublishedPublicly = options.availability.isAvailable && !options.availability.isUnknown && options.runPublish && (pkg.publishConfig && pkg.publishConfig.access !== 'restricted') && !npm.isExternalRegistry(pkg);
+
 	const answers = await inquirer.prompt({
 		version: {
 			type: 'list',
-			message: 'Select semver increment or specify new version',
-			pageSize: Version.SEMVER_INCREMENTS.length + 2,
-			choices: [...Version.SEMVER_INCREMENTS
-				.map(inc => ({
-					name: `${inc} 	${prettyVersionDiff(oldVersion, inc)}`,
+			message: 'Select SemVer increment or specify new version',
+			pageSize: SEMVER_INCREMENTS.length + 2,
+			choices: [
+				...SEMVER_INCREMENTS.map(inc => ({ // TODO: prerelease prefix here too
+					name: `${inc} 	${new Version(oldVersion, inc).format()}`,
 					value: inc,
 				})),
-			new inquirer.Separator(),
-			{
-				name: 'Other (specify)',
-				value: null,
-			}],
-			filter: input => Version.isValidInput(input) ? new Version(oldVersion).getNewVersionFrom(input) : input,
+				new inquirer.Separator(),
+				{
+					name: 'Other (specify)',
+					value: undefined,
+				},
+			],
+			filter: input => input ? new Version(oldVersion, input) : input,
 		},
 		customVersion: {
 			type: 'input',
 			message: 'Version',
-			when: answers => !answers.version,
-			filter: input => Version.isValidInput(input) ? new Version(pkg.version).getNewVersionFrom(input) : input,
-			validate(input) {
-				if (!Version.isValidInput(input)) {
-					return 'Please specify a valid semver, for example, `1.2.3`. See https://semver.org';
+			when: answers => answers.version === undefined,
+			filter(input) {
+				if (SEMVER_INCREMENTS.includes(input)) {
+					throw new Error('Custom version should not be a `SemVer` increment.');
```

Comment:
```suggestion
					throw new Error('Custom version should not be a SemVer increment.');
```

Same with other places.